source/extensions/filters/http/common/aws/signer_impl.cc

```diff
@@ -0,0 +1,145 @@
+#include "extensions/filters/http/common/aws/signer_impl.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/http/headers.h"
+#include "common/ssl/utility.h"
+
+#include "extensions/filters/http/common/aws/utility.h"
+
+#include "absl/strings/str_join.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+void SignerImpl::sign(Http::Message& message) {
```

Comment:
This whole function assumes that the complete request body is available at this point. I'm not sure how are you going to use this, but Envoy doesn't buffer requests before forwarding them, so this won't work for proxied requests.

Reply:
That is a good point. In the sigv4 algorithm, the request body signing is technically optional (for this very reason), but adds an extra layer of security for certain types of requests. I'll add a parameter, much like we do in the official SDK to control signing the body. The caveat being, that the body must be fully available at that point.